compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -280,6 +301,16 @@ private boolean isAssignable(BType source, BType target, List<TypePair> unresolv
             return true;
         }
 
+        // Check whether the source is a proper sub set of the target.
+        if (source.tag == TypeTags.FINITE && target.tag == TypeTags.FINITE) {
+            BFiniteType finiteType = (BFiniteType) source;
+            boolean isAssignable = finiteType.valueSpace.stream()
```

Comment:
shouldn't we just return the result here? (false or true) with this logic, it flows through if it is false,